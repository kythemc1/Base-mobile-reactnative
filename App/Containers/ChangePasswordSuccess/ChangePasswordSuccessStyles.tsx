import { StyleSheet } from 'react-native';
import sizes from 'Configs/UI/Sizes';
import {colors} from 'Configs/UI/Colors';

export const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: colors.bgNoti
    },

    header: {
        flex: 1,
        marginTop: sizes._screenHeight * 0.04,
    },
    then: {
        flex: 9,
    },


    text: {
        alignSelf: 'center',
        color: 'white',
        fontFamily: 'LexendDeca-Light',
    },


    Wc: {
        alignSelf: 'center',
        width: sizes._screenWidth * 0.5,
        height: sizes._screenWidth * 0.5,
        marginTop: sizes._screenHeight * 0.08,
    },
    textwc: {
        textAlign: 'center',
        fontFamily: 'LexendDeca-Medium',
        fontSize: sizes._25sdp,
        color: colors.color_primary1,
    },

    body: {},
    endThen: { alignItems: 'center',top: sizes._8sdp},
    textInstructions:{
        color: colors.textunder,
        alignSelf: 'center',
        fontSize: sizes._12sdp,
        fontFamily: 'LexendDeca-Light'
    },
    wraplogin:{
        alignSelf: 'center',
        top : sizes._40sdp
    }
});
