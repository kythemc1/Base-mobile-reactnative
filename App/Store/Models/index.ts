export interface IAuthState {
    isLogged: boolean;
    token: string | null;
    refreshToken: string | null;
    refreshTokenExpiredDate: number | null;
    type: string | null;
}

export interface IUserState {
  id: number | null;
  address: string | null;
  taxCodeCompany: string | null;
  avatar: string | null;
  birthday: Date | null;
  firstName: string | null;
  gender: string | null;
  email: string | null;
  fullNameUser: string | null;
  status: string | null;
  username: string | null;
  showAlertRecovery: boolean | null;
  showAlertSignIn: boolean | null;
  showAlertSignIn5: boolean | null;
  user_keycloak_id: string | null;
  showPasswordnotmatch: boolean | null;
  showPassworderror: boolean | null;
}

export interface GetCustomerInterface {
  id: string | null;
  name: string | null;
  certificateNo: string | null;
  taxCode: string | null;
  certificateOrTaxCode: string | null;
  type: string | null;
  address: string | null;
  email: string | null;
  phone: string | null;
  userCreatedName: string | null;
}

export interface ListCustomerInterface {
  idCustomer: string | null;
  nameCustomer: string | null;
  numberPapersCustomer: string | null;
  typeCustomer: string | null;
  adressCustomer: string | null;
  emailCustomer: string | null;
  phoneNumberCustomer: string | null;
  createPerson: string | null;
}

export interface Filter {
  page: number | null;
  size: number | null;
  sorts: {} | undefined;
  customerType?: string | null;
  email?: string | null;
  phone?: string | null;
  keyword?: string | null;
  sort?: string | null;
  usersCreated: any;
}

export interface ParamsCreator {
  filter: string | null;
  page: number | null;
  size: number | null;
}

export interface Creator {
  isLoading: boolean | undefined;
  error: string | null;
  listCreator: [] | undefined;
  creator: any;
  valueCreator: [] | undefined;
  valueDropdownCreator: string | null;
  isFocusCreator: boolean | undefined;
}

export interface CustomerInterface {
  customer: any;
  listCustomers: [];
  valueCustomer: any;
  isLoading: boolean;
  error: string;
  valueDropdownCustomer: string;
  isFocusCustomer: boolean;
  totalPages: number;
  listSearchCustomer: [];
  loadingPagination: boolean;
}

export interface GetCreatorInterface {
  userId: string | null;
  username: string | null;
  email: string | null;
  fullName: string | null;
  phoneNumber: string | null;
  type: string | null;
  branchId: string | null;
  companyId: string | null;
  position: null;
  mainDepartment: null;
  orderDepartment: null;
  orderUser: null;
}

export interface NotificationInterface {
  type: string;
  content: string;
  userId: string;
  notificationId: string;
  title: string;
  dataId: any | null;
  hasRead: boolean;
  createTime: string;
}

export interface InitialNotificationInterface {
  listNotification: [];
  isLoading: boolean;
  isError: string;
  deviceId: string;
  uniqueId: string;
}

export interface ParamsNotification {
  page: number;
  size: number;
  sorts: {};
}

//  models document
export interface ParamsMyDocument {
  page: number;
  size: number;
  status?: string;
  sorts?: any;
  keyword?: string;
  documentTypeIds?: any;
  dateTo?: string;
  dateFrom?: string;
}

export interface MyDocumentInterface {
  documentId: string;
  name: string;
  code: string;
  documentTypeName: string;
  emailPartner: string;
  namePartner: string;
  createdDate: string;
  status: string;
}

export interface ListMyDocumentInterface {
  contractingParties: string;
  createdDate: string;
  documentCode: string;
  documentId: string;
  documentName: string;
  documentTypeName: string;
  emailPartner: string;
  status: number;
  imgDocument: string;
}

/**NOTE**/
// documentType Interface
export interface DocumentTypeInterface {
  documentTypeId: string;
  code: string;
  name: string;
  description: string;
  companyId: string;
  status: string;
  version: number;
}
export interface AmountTextInterface {
    processingDocumentNumber : number,
    expireDocumentNumber: number,
    doneDocumentNumber: number,
    manageDocumentNumber: number,

}

export interface localUser {
  userName: string,
  password: string
}
