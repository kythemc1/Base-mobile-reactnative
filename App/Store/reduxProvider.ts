import {configureStore} from '@reduxjs/toolkit';
import {combineReducers} from 'redux';
import {
    FLUSH,
    PAUSE,
    PERSIST,
    persistReducer,
    persistStore,
    PURGE,
    REGISTER,
    REHYDRATE,
} from 'redux-persist';
import {generatePersistConfig} from 'Utils/Redux';
import authSlice from './Reducers/authSlice';
import langSlice from './Reducers/langSlice';
import customerSlice from './Reducers/customerSlice';
import creatorSlice from 'Store/Reducers/creatorSlice';
import notificationSlice from 'Store/Reducers/notificationSlice';
import myDocumentSlice from 'Store/Reducers/myDocumentSlice';
import DocumentTypeSlice from 'Store/Reducers/documentTypeSlice';
import amountText from 'Store/Reducers/amoutTextSlice';
import documentCompanySlice from './Reducers/documentCompanySlice';

const reducers = combineReducers({
    auth: authSlice,
    lang: langSlice,
    customer: customerSlice,
    creator: creatorSlice,
    notification: notificationSlice,
    myDocument: myDocumentSlice,
    documentType: DocumentTypeSlice,
    amountText: amountText,
    documentCompany: documentCompanySlice,
});

const version = 22021401;

const persistedReducer = persistReducer(
    generatePersistConfig({
        whitelist: [
            'auth',
            'lang',
            'customer',
            'creator',
            'notification',
            'myDocument',
            'amountText',
            'documentType',
            'documentCompany',
        ],
        version,
    }),
    reducers,
);

const store = configureStore({
    reducer: persistedReducer,
    middleware: getDefaultMiddleware => {
        const middlewares = getDefaultMiddleware({
            serializableCheck: {
                ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
            },
        });

        return middlewares;
    },
});

const persistor = persistStore(store);

export {store, persistor};

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
