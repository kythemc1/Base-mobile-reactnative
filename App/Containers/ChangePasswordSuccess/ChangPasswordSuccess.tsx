import React from 'react';
import {
    View,
    Image,
    SafeAreaView,
    Text
} from 'react-native';
import {useSelector} from 'react-redux';
import {useTranslation} from 'react-i18next';
import {useChangeLang} from '../../Hooks/useLang';
import {RootState} from '../../Store/reduxProvider';
import {HeaderOnboard} from 'Components/Commons/HeaderOnboard';
import images from 'Configs/Constants/images';
import {Button} from 'Components/Commons/Button';
import {styles} from 'Containers/ChangePasswordSuccess/ChangePasswordSuccessStyles';

export default function BackLogin({navigation}: any) {
    const {t} = useTranslation();
    const {lang} = useSelector((state: RootState) => state.lang);
    return (
        <SafeAreaView style={styles.container}>
            <View style={styles.header}>
                <HeaderOnboard></HeaderOnboard>
            </View>
            <View style={styles.then}>

                <View style={styles.body}>
                    <Image
                        style={styles.Wc}
                        source={images.success_img}
                    />
                </View>
                <Text style={styles.textwc}>{t('changepasssuccess.changepasssuccess')}</Text>
                <View style={styles.endThen}>
                    <Text style={styles.textInstructions}>
                        {t('backlogin.instructions')}
                    </Text>
                    <Text
                        style={styles.textInstructions}>
                        {t('changepasssuccess.alertline1')}
                    </Text>
                    <Text style={styles.textInstructions}>
                        {t('changepasssuccess.alertline2')}
                    </Text>

                </View>

                <View style={styles.wraplogin}>
                    <Button sizeButton={'large'} onPressButton={() => navigation.navigate('SignIn')}
                        text={`${t('sign_in.sign_in')}`} typeButton={'primary'}></Button>
                </View>
            </View>

        </SafeAreaView>
    );
}



