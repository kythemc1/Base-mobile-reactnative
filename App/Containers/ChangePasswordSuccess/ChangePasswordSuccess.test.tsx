import renderer from 'react-test-renderer';
import React from 'react';
import ChangePasswordSuccess from './ChangPasswordSuccess';
import {Provider} from 'react-redux';
import {store} from 'Store/reduxProvider';
//Fix Invariant Violation: `new NativeEventEmitter()` requires a non-null argument.
jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter');
// mock moduleName react-native-firebase
jest.mock('@react-native-firebase/messaging', () => ({
    messaging: jest.fn(() => ({
        hasPermission: jest.fn(() => Promise.resolve(true)),
        subscribeToTopic: jest.fn(),
        unsubscribeFromTopic: jest.fn(),
        requestPermission: jest.fn(() => Promise.resolve(true)),
        getToken: jest.fn(() => Promise.resolve('myMockToken')),
    })),
    notifications: jest.fn(() => ({
        onNotification: jest.fn(),
        onNotificationDisplayed: jest.fn(),
    })),
    analytics: jest.fn(() => ({
        logEvent: jest.fn(),
        setUserProperties: jest.fn(),
        setUserId: jest.fn(),
        setCurrentScreen: jest.fn(),
    })),
}));
test('renders correctly', () => {
    const tree = renderer
        .create(
            <Provider store={store}>
                <ChangePasswordSuccess />
            </Provider>,
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
jest.mock('react-i18next', () => ({
    useTranslation: () => ({
        t: (str: string) => str,
        i18n: {
            changeLanguage: () => new Promise(() => {}),
        },
    }),

    initReactI18next: {
        type: '3rdParty',
        init: jest.fn(),
    },
}));
