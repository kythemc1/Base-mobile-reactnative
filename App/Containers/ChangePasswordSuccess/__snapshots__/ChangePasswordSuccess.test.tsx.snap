// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
>
  <BVLinearGradient
    colors={
      [
        4294967295,
        4294967295,
        4294967295,
        4294967295,
        4292205539,
      ]
    }
    endPoint={
      {
        "x": 0,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      {
        "x": 0,
        "y": 0,
      }
    }
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "flex": 1,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          [
            {
              "flex": 1,
            },
            {
              "paddingBottom": 0,
            },
          ]
        }
      >
        <View
          style={
            {
              "flex": 0.2,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginLeft": 37.5,
                  "marginTop": 71.25,
                  "width": 150,
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
              >
                <Image
                  source={
                    {
                      "testUri": "../../../App/Assets/Login/BackIcon.png",
                    }
                  }
                  style={
                    {
                      "height": 75,
                      "width": 75,
                    }
                  }
                />
              </View>
              <View>
                <Text
                  style={
                    {
                      "color": "rgba(20, 57, 128, 1)",
                      "fontFamily": "LexendDeca-Light",
                      "fontSize": 15,
                      "marginTop": 8,
                    }
                  }
                >
                  sign_in.back
                </Text>
              </View>
            </View>
            <View
              style={
                {
                  "marginLeft": 390,
                  "marginTop": 71.25,
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "flexDirection": "row",
                    "opacity": 1,
                  }
                }
              >
                <Image
                  source={
                    {
                      "testUri": "../../../App/Assets/SplashSlider/Vi.png",
                    }
                  }
                  style={
                    {
                      "height": 70.3125,
                      "width": 112.5,
                    }
                  }
                />
                <Text
                  style={
                    {
                      "color": "rgba(27, 61, 118, 1)",
                      "fontFamily": "LexendDeca-Medium",
                      "fontSize": 20,
                      "marginLeft": 0,
                      "marginTop": 4,
                    }
                  }
                >
                  language.changeLanguage
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "flex": 0.45,
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../App/Assets/SplashSlider/Sidoc.png",
              }
            }
            style={
              {
                "alignSelf": "center",
                "height": 150,
                "marginTop": 106.72,
                "width": 375,
              }
            }
          />
          <Text
            style={
              {
                "alignSelf": "center",
                "color": "rgba(130, 130, 130, 1)",
                "fontSize": 13.5,
              }
            }
          >
            Document System Electronic
          </Text>
        </View>
        <Text
          style={
            {
              "color": "#143980",
              "flex": 0.15,
              "fontFamily": "LexendDeca-Medium",
              "fontSize": 25,
              "marginTop": -37.5,
              "textAlign": "center",
            }
          }
        >
          changepasssuccess.changepasssuccess
        </Text>
        <View
          style={
            {
              "flex": 0.36,
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../App/Assets/ChangePassword/ChangePassSuccess.png",
              }
            }
            style={
              {
                "alignSelf": "center",
                "height": 135,
                "width": 112.5,
              }
            }
          />
          <View
            style={
              {
                "alignItems": "center",
                "marginTop": 30,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#0E1E38",
                  "fontFamily": "LexendDeca-Light",
                  "fontSize": 18,
                }
              }
            >
              changepasssuccess.alertline1
            </Text>
            <Text
              style={
                {
                  "color": "#0E1E38",
                  "fontFamily": "LexendDeca-Light",
                  "fontSize": 18,
                }
              }
            >
              changepasssuccess.alertline2
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "alignSelf": "center",
            "height": 320.15999999999997,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#143980",
              "borderRadius": 10,
              "marginTop": 146.74,
              "opacity": 1,
              "paddingHorizontal": 32,
              "paddingVertical": 12,
              "width": 262.5,
            }
          }
        >
          <Text
            style={
              {
                "alignSelf": "center",
                "color": "white",
                "fontFamily": "LexendDeca-Light",
              }
            }
          >
            changepasssuccess.button
          </Text>
        </View>
      </View>
    </View>
  </BVLinearGradient>
</RCTSafeAreaView>
`;
